components/script/script_thread.rs

```diff
@@ -869,6 +867,52 @@ impl ScriptThread {
         });
     }
 
+    /// Step 13 of https://html.spec.whatwg.org/multipage/#navigate
+    pub fn navigate(parent_pipeline_id: PipelineId, mut load_data: LoadData, replace: bool) {
+        SCRIPT_THREAD_ROOT.with(|root| {
+            let script_thread = match root.get() {
+                None => return,
+                Some(script) => script,
+            };
+            let script_thread = unsafe { &*script_thread };
+            let is_javascript = load_data.url.scheme() == "javascript";
+            // If resource is a request whose url's scheme is "javascript"
```

Comment:
Spec link? to https://html.spec.whatwg.org/multipage/browsing-the-web.html#javascript-protocol